### PR TITLE
WT-7706 Prepared transaction abort should restore correct update from the history store.

### PR DESCRIPTION
**Pull request by hbokhari**

### CODE REVIEW

File: src/reconcile/rec_visibility.c [hunk 1]
```diff
@@ -517,6 +517,17 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
...
                     WT_ASSERT(session, same_txn_valid_upd->type != WT_UPDATE_TOMBSTONE);
                     upd_select->upd = upd = same_txn_valid_upd;
+
+                } else if (same_txn_valid_upd != NULL && vpack != NULL && vpack->tw.prepare) {
+                    /*
+                     * The on-disk version is from an aborted prepare transaction. Therefore, use
+                     * the update from the same transaction as the selected update.
+                     */
+                    WT_ASSERT(session,
+                      first_upd->prepare_state == WT_PREPARE_INPROGRESS &&
```
Comment (quchenhao) [inline src/reconcile/rec_visibility.c:527]: I think it is not correct to assert the first_upd is the prepared update. It is possible that there are more aborted updates in front of it, such as a reserved update then aborted.

**Review by kommiharibabu — APPROVED** [review]
lgtm

**Review by quchenhao — APPROVED** [review]
lgtm

**Review by kommiharibabu — APPROVED** [review]
lgtm

**Pull request merged**